dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,679 @@
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from 0xMETADATAVIEWSADDRESS
+
+// The DSSCollection contract
+//
+pub contract DSSCollection: NonFungibleToken {
+
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        description: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event ItemCreated(
+        id: UInt64,
+        itemID: UInt64,
+        points: UInt64,
+        itemType: String
+    )
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String
+    )
+    pub event ItemAddedToSlot(
+        id: UInt64,
+        slotID: UInt64,
+        collectionGroupID: UInt64
+    )
+    pub event CollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt8
+    )
+    pub event DSSCollectionNFTBurned(id: UInt64)
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:                 UInt64
+
+    // Lists in contract
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+    access(self) let slotByID: @{UInt64: Slot}
+    access(self) let itemByID: @{UInt64: Item}
+
+    // A public struct to access Slot data
+    //
+    pub struct ItemData {
+        pub let id: UInt64 // unique uuid of resource
+        pub let itemID: UInt64 // the id of the edition, tier, play
+        pub let points: UInt64 // points for item
+        pub let itemType: String // (edition.id, edition.tier, play.id)
+
+        init (id: UInt64) {
+            if let item = &DSSCollection.itemByID[id] as &DSSCollection.Item? {
+                self.id = item.id
+                self.itemID = item.itemID
+                self.points = item.points
+                self.itemType = item.itemType
+            } else {
+                panic("item does not exist")
+            }
+        }
+    }
+
+    // A top-level Item with a unique ID
+    //
+    pub resource Item {
+        pub let id: UInt64 // unique uuid of resource
+        pub let itemID: UInt64 // the id of the edition, tier, play
+        pub let points: UInt64 // points for item
+        pub let itemType: String // (edition.id, edition.tier, play.id)
```

Comment:
a `slot_id` could be helpful in navigating from an item.

Reply:
Ah. An Item only has a slot when it is added to the slot. You can get the slotId in the ItemAddedToSlot event.